File: Java/Buku.java
public class Buku extends Produk {
    private String penulis;
    private String penerbit;
    private int tahunTerbit;

    public Buku(String kodeProduk, String namaProduk, int harga,
                String penulis, String penerbit, int tahunTerbit) {
        super(kodeProduk, namaProduk, harga);
        this.penulis = penulis;
        this.penerbit = penerbit;
        this.tahunTerbit = tahunTerbit;
    }

    public String getPenulis() { return penulis; }
    public String getPenerbit() { return penerbit; }
    public int getTahunTerbit() { return tahunTerbit; }

    public void setPenulis(String penulis) { this.penulis = penulis; }
    public void setPenerbit(String penerbit) { this.penerbit = penerbit; }
    public void setTahunTerbit(int tahunTerbit) { this.tahunTerbit = tahunTerbit; }
}

File: Python/Buku.py
from Produk import Produk

class Buku(Produk):
    def __init__(self, kodeProduk, namaProduk, harga, penulis, penerbit, tahunTerbit):
        super().__init__(kodeProduk, namaProduk, harga)
        self.penulis = penulis
        self.penerbit = penerbit
        self.tahunTerbit = tahunTerbit

    def getPenulis(self):
        return self.penulis

    def getPenerbit(self):
        return self.penerbit

    def getTahunTerbit(self):
        return self.tahunTerbit

    def setPenulis(self, penulis):
        self.penulis = penulis

    def setPenerbit(self, penerbit):
        self.penerbit = penerbit

    def setTahunTerbit(self, tahunTerbit):
        self.tahunTerbit = tahunTerbit
